resolversrv/utils.py

```diff
@@ -92,10 +157,45 @@ def add_records(datalinks_records_list):
                 session.execute(on_conflict_stmt)
             current_app.logger.info('updated db with new data successfully')
             return True, 'updated db with new data successfully'
-        except (SQLAlchemyError, DBAPIError) as e:
+        except SQLAlchemyError as e:
             current_app.logger.error('SQLAlchemy: ' + str(e))
             return False, 'SQLAlchemy: ' + str(e)
-    return False, 'unable to extract data from protobuf structure'
+    return False, 'unable to add records to the database'
+
+
+def add_records_new(documents):
+    """
+    upserts records into db
+
+    :param documents:
+    :return: success boolean, plus a status text for retuning error message, if any, to the calling program
+    """
+    rows = []
+    for doc in documents.document_records:
+        rows.append({"bibcode":doc.bibcode,
+                     "identifier": doc.identifier,
+                     "links": json_format.MessageToDict(doc.links)})
+
+    if len(rows) > 0:
+        table = Documents.__table__
```

Comment:
if you say you are trying to shave off milliseconds by executing SQL statements `session.execute` (as opposed to ORM like everywhere else) then you should not do `current_app.logger.info` because you are going to lose milliseconds there.

the `session.execute` is justified if important savings can be had -- it makes the code db engine specific (which isn't portable nor consistent)

so you should pick one; by measuring if `session.execute` really gives you the perceived benefit. If not, then stick to the ORM form

Reply:
done.